src/Aspire.Hosting/DistributedApplicationExecutionContextOptions.cs

```diff
@@ -12,16 +12,17 @@ public class DistributedApplicationExecutionContextOptions
     /// Constructs a <see cref="DistributedApplicationExecutionContextOptions" />.
     /// </summary>
     /// <param name="operation">Indicates whether the AppHost is running in Publish mode or Run mode.</param>
-    public DistributedApplicationExecutionContextOptions(DistributedApplicationOperation operation) : this(operation, null)
+    public DistributedApplicationExecutionContextOptions(DistributedApplicationOperation operation)
     {
+        this.Operation = operation;
     }
 
     /// <summary>
     /// Constructs a <see cref="DistributedApplicationExecutionContextOptions" />.
     /// </summary>
     /// <param name="operation">Indicates whether the AppHost is running in Publish mode or Run mode.</param>
     /// <param name="publisherName">The publisher name if in Publish mode.</param>
-    public DistributedApplicationExecutionContextOptions(DistributedApplicationOperation operation, string? publisherName = null)
+    public DistributedApplicationExecutionContextOptions(DistributedApplicationOperation operation, string publisherName)
```

Comment:
Why couldn't this stay as

```csharp
public DistributedApplicationExecutionContextOptions(DistributedApplicationOperation operation, string? publisherName)
```

Reply:
Had a call with @mitchdenny and if you want to no set a valid publishname you can use the other ctor.

Reply:
In general, that kind of API pattern makes using the API harder to use. It maybe doesn't matter much for this API, but in general, the pattern isn't great.

```csharp
void Foo() { ... }
void Foo(string v) { ... }

// consume:

string? someV = GetSomeV();

if (someV is null)
{
    Foo();
}
else
{
    Foo(someV);
}
```

vs.

```csharp
void Foo() { ... }
void Foo(string? v) { ... }

// consume:

string? someV = GetSomeV();
Foo(someV);
```

Reply:
Got it. In that case the usage is based on the first argument, `operation`, and if you are creating a `Publish` operation you have to set a non-null `publisherName`. So maybe the API could have been more specific about it, even maybe preventing from setting a `publisherName` if you create another type (`Run` or `Inspect`). For instance `ctor(string publisherName)` would be more adapted, it would infer the `Publish` operation. Or maybe these should be sub-classes.